src/models/deployers/PresaleTemplateDeployer.js

```diff
@@ -0,0 +1,212 @@
+const { APP_IDS } = require('@aragon/templates-externally-owned-presale-bonding-curve/lib/helpers/constants.js')
+const BaseDeployer = require('./BaseDeployer')
+const logger = require('../../helpers/logger')('PresaleTemplateDeployer')
+const { getInstalledApps, getInstalledAppsById } = require('../../helpers/events')
+
+const VERIFICATION_HEADERS = [
+  'Commit sha: 68b894fe4fd81cbf43c8b1e128be9ddfffd41380',
+  'GitHub repository: https://github.com/aragonone/fundraising',
+  'Tool used for the deploy: https://github.com/aragon/aragon-network-deploy',
+]
+
+const ZERO_ADDRESS = '0x' + '0'.repeat(40)
+
+module.exports = class extends BaseDeployer {
+  constructor(config, environment, output, verifier = undefined) {
+    super(environment, output, verifier, logger)
+    this.config = config
+  }
+
+  async call() {
+    const configTemplate = this.previousDeploy.presaleTemplate
+    const PresaleTemplate = await this.environment.getArtifact('EOPBCTemplate', '@aragon/templates-externally-owned-presale-bonding-curve')
+
+    if (configTemplate && configTemplate.address) {
+      await this._loadPresaleTemplate(PresaleTemplate, configTemplate.address)
+    } else {
+      await this._deployPresaleTemplate(PresaleTemplate)
+    }
+
+    await this._verifyPresaleTemplate()
+
+    await this._deployPresaleInstance()
+    await this._verifyFundraisingContracts()
+  }
+
+  // ************ Template ************* //
+
+  async _loadPresaleTemplate(PresaleTemplate, templateAddress) {
+    logger.warn(`Using previous deployed Presale Template at ${templateAddress}`)
+    this.presaleTemplate = await PresaleTemplate.at(templateAddress)
+  }
+
+  async _deployPresaleTemplate(PresaleTemplate) {
+    this._printPresaleTemplateDeploy()
+    const { ens, daoFactory, registrar } = this.config.infra
+
+    this.presaleTemplate = await PresaleTemplate.new(daoFactory, ens, registrar)
+
+    const { address, transactionHash } = this.presaleTemplate
+    logger.success(`Created Presale template contract at ${address}`)
+    this._saveDeploy({ presaleTemplate: { address, transactionHash }})
+  }
+
+  _printPresaleTemplateDeploy() {
+    logger.info('Deploying Presale Template contract with config:')
+    logger.info(` - DAO factory:                             ${this.config.infra.daoFactory}`)
+    logger.info(` - ENS:                                     ${this.config.infra.ens}`)
+    logger.info(` - Registrar:                               ${this.config.infra.registrar}`)
+  }
+
+  async _verifyPresaleTemplate() {
+    await this._verify('presaleTemplate')
+  }
+
+  // ************ Instance ************* //
+
+  async _deployPresaleInstance() {
```

Comment:
We shouldn't do this in the deployment script since it is not going to work on mainnet. The script should just deploy the template, but not instantiate it cause it will need to be called from the AN DAO agent app.

We still need to work on an EVM script that basically does what you are doing here, transferring the token controller and calling the template.

Reply:
Why won't it work on mainnet? Because of the token controller? Who is it right now? The AN Dao? Can't we then just change it before to the address running these scripts?
Anyway, what I did is to add a parameter `--instance` so it doesn't deploy the instance by default (unless you specify that parameter).